### PR TITLE
Xml format maven plugin, for application context and assembly descriptors

### PR DESCRIPTION
**Pull request by aaime**
Applies the [acegi xml-format-maven-plugin](https://github.com/acegi/xml-format-maven-plugin) to Spring application context files and assembly descritptors.

The change is a draft, not to be merged at present, because it's using a modified version of the plugin, [based on this pull request](https://github.com/acegi/xml-format-maven-plugin/pull/33), which adds an option to preserve empty lines, with a behavior similar to sortpom. This PR depends on a SNAPSHOT version of xml-format-maven-plugin, not deployed anywhere, as such, all the PR checks are failing.

The PR is meant to share the results of the formatting and discuss. @mprins think you might be interested into it :-D

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/community/backup-restore/core/src/test/resources/applicationContext.xml [hunk 1]
```diff
@@ -1,28 +1,19 @@
...
-	xmlns:util="http://www.springframework.org/schema/util"
-	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xsi:schemaLocation="
-		http://www.springframework.org/schema/beans
-        http://www.springframework.org/schema/beans/spring-beans-4.0.xsd
-        http://www.springframework.org/schema/batch
-        http://www.springframework.org/schema/batch/spring-batch-3.0.xsd
-        http://www.springframework.org/schema/util
-        http://www.springframework.org/schema/util/spring-util-4.0.xsd">
+<beans xmlns="http://www.springframework.org/schema/beans" xmlns:batch="http://www.springframework.org/schema/batch" xmlns:util="http://www.springframework.org/schema/util" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="   http://www.springframework.org/schema/beans         http://www.springframework.org/schema/beans/spring-beans-4.0.xsd         http://www.springframework.org/schema/batch         http://www.springframework.org/schema/batch/spring-batch-3.0.xsd         http://www.springframework.org/schema/util         http://www.springframework.org/schema/util/spring-util-4.0.xsd">
```
Comment (aaime) [inline src/community/backup-restore/core/src/test/resources/applicationContext.xml:9]: One thing that is bothering me some, is that the elements are kept on a single line, despite the number of attributes they might contain. However, the plugin has no option for line wrapping. That said, I still prefer to output of the reforat to the current status quo :-D

**Review by bradh — COMMENTED** [review]
Other than wondering why we sometimes have DOCTYPE, and sometimes not, generally better.

Line wrapping would make it perfect.

File: src/community/backup-restore/core/src/main/resources/applicationContext.xml [hunk 2]
```diff
@@ -1,852 +1,851 @@
 <?xml version="1.0" encoding="UTF-8"?>
+
 <!-- <!DOCTYPE beans PUBLIC "-//SPRING//DTD BEAN//EN" "http://www.springframework.org/dtd/spring-beans.dtd"> -->
 <!-- 
  (c) 2016 Open Source Geospatial Foundation - all rights reserved
  This code is licensed under the GPL 2.0 license, available at the root
  application directory. 
 -->
+<beans xmlns="http://www.springframework.org/schema/beans" xmlns:batch="http://www.springframework.org/schema/batch" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.springframework.org/schema/beans         http://www.springframework.org/schema/beans/spring-beans-4.0.xsd         http://www.springframework.org/schema/batch         http://www.springframework.org/schema/batch/spring-batch-3.0.xsd">
```
[the remark this answers is not in the document]
Comment (mprins) [reply src/community/backup-restore/core/src/main/resources/applicationContext.xml:9]: yeah, but these are unlikely to change much. 
It seems odd though that there are a bunch of spaces between the various schemaLocation urls

File: src/community/backup-restore/core/src/test/resources/applicationContext.xml (hunk 1 above)
Comment (mprins) [reply src/community/backup-restore/core/src/test/resources/applicationContext.xml:9]: yes. 
I find the many extra whitespaces in the schemaLocation a bit odd, perhaps those can be fixed with a second run

**Comment by mprins** [comment]
(tangentially related) Having all the `applicationContext.xml` together I notice there's a medley of both XML Schema and DTD. I think it would be good to go with schema for all of those - that would also make it possible to validate them

File: src/community/jms-cluster/activemqBroker/src/main/webapp/WEB-INF/classes/applicationContext.xml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
+
 <!-- <!DOCTYPE beans PUBLIC "-//ACTIVEMQ//DTD//EN" "http://activemq.org/dtd/activemq.dtd">
```
[the remark this answers is not in the document]
Comment (aaime) [reply src/community/jms-cluster/activemqBroker/src/main/webapp/WEB-INF/classes/applicationContext.xml:3]: Good catch! I don't know the answer though :-D

[the remark this answers is not in the document]
Comment (mprins) [reply src/community/jms-cluster/activemqBroker/src/main/webapp/WEB-INF/classes/applicationContext.xml:3]: `<beans ...` has schema defined so yes it should be commented out - in fact the `<DOCTYPE... ` should be removed (see https://osgeo-org.atlassian.net/browse/GEOS-10029)

**Comment by aaime** [comment]
Still want to work on it.

**Comment by aaime** [comment]
So the main issue here is that the formatter won't wrap the attributes, which causes the first element, with lots of namespace declaration, to be one giant line... this would require a further set of changes to xml-format-maven-plugin, with further overrides compared to the normal behavior of dom4j.

**Pull request closed without merging**